Improve the multi-node HA section (#601)
timescale/docs: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md

> # High availability (HA) with multi-node

Reviewer: ```suggestion
# High availability with multi-node
```

A multi-node installation of TimescaleDB can be made highly available
in two different ways:

1. by setting up one or more standbys for each node in the cluster, or
2. by natively replicating data at the chunk level.

The first option relies on streaming replication and the setup is
similar to [configuring single-node HA][single-ha], although the
configuration needs to be applied to each node independently.

> The second option relies on built-in capabilities of multi-node
> TimescaleDB to replicate data at the chunk level instead of
> replicating entire data nodes. The access node still relies on a
> streaming replication standby, but the data nodes need no additional
> configuration. Instead, the existing pool of data nodes share
> responsibility to host chunk replicas in order to be able to sustain
> node failures.

Reviewer: ```suggestion
To replicate data at the chunk level, you can use the built-in 
capabilities of multi-node TimescaleDB to avoid having to
replicate entire data nodes. The access node still relies on a
streaming replication standby, but the data nodes need no additional
configuration. Instead, the existing pool of data nodes share
responsibility to host chunk replicas and handle node failures.
```

> There are pros and cons of each approach. Setting up standbys for each
> node in the cluster ensures that standbys are identical at the
> instance level and is a tried and tested method to provide high
> availability. However, it is also a heavy-weight solution and
> essentially requires setting up a mirror cluster.

Reviewer: ```suggestion
There are advantages and disadvantages to each approach. 
Setting up standbys for each node in the cluster ensures that 
standbys are identical at the instance level, and this is a tried
and tested method to provide high availability. However, it also
requires more setting up and maintenance for the mirror cluster.
```

Author: "it also requires more setting up" --- sounds a bit weird to me, but I am not a native English speaker. Just want to double-check @Loquacity

> Native replication, on the other hand, offers simpler configuration
> and additional, built-in, capabilities, like being able to easier add
> and remove data nodes, and use different replication factors on each
> distributed hypertable. The downside, however, is that only chunks are
> replicated on the data nodes.

Reviewer: ```suggestion
Native replication requires less configuration, and takes advantage of 
built-in capabilities, such as adding and removing data nodes, and 
different replication factors on each distributed hypertable. However, 
only chunks are replicated on the data nodes.
```

> The rest of this section focuses on native replication. To setup
> standbys for each node, follow the instructions for [single node
> HA][single-ha].

Reviewer: ```suggestion
The rest of this section discusses native replication. To set up
standbys for each node, follow the instructions for [single node
HA][single-ha].
```

> ## Native replication
> 
> Native replication is a set of capabilities and APIs around which one
> can build a highly available multi-node TimescaleDB installation. At
> the core of native replication is the ability to write copies
> (replicas) of a chunk to multiple data nodes. Thus, a failure of one
> data node should not lead to loss of data as copies exist on other
> data nodes. If a data node is permanently lost, a new data node can be
> added to the cluster, and lost chunk replicas can be copied
> (re-replicated) from other data nodes to again reach the number of
> desired chunk replicas.

Reviewer: ```suggestion
## Native replication
Native replication is a set of capabilities and APIs that allow you to 
build a highly available multi-node TimescaleDB installation. At
the core of native replication is the ability to write copies or replicas
of a chunk to multiple data nodes. If one data node fails, copies exist
on other data nodes, so that data is not lost. If a data node is 
permanently lost, a new data node can be added to the cluster, 
and lost chunk replicas can be copied or re-replicated from 
other data nodes to reach the number of desired chunk replicas.
```

> Note that orchestration around native replication is not provided as
> part of TimescaleDB itself, so, like standard single-node standby
> setups, external entities need to manage the set of data nodes,
> trigger failover, and re-replicate chunks.

Reviewer: ```suggestion
Native replication is not provided in TimescaleDB by default. You can
set this up in the same way as a standard single-node standby, and
use external entities to manage the set of data nodes, trigger failover, 
and re-replicate chunks. This section contains information about how
to do that.
```

Author: I think the suggested changes actually subtly changed the meaning of what I was trying to say and is no longer entirely correct. I'll try to tweak my original text to match your suggestion while trying to preserve the original messaging.

> <highlight type="warning"> Native replication is currently under
> development and lacks functionality for a complete high-availability
> solution. Some functionality described in this section is still
> experimental. For production environments, we recommend setting up
> standbys for each node in a multi-node cluster.</highlight>

Reviewer: ```suggestion
<highlight type="warning"> 
Native replication in TimescaleDB is under development and 
currently lacks functionality for a complete high-availability
solution. Some functionality described in this section is still
experimental. For production environments, we recommend setting up
standbys for each node in a multi-node cluster.
</highlight>
```

### Configuring native replication

The first step to enable native replication is to configure a standby
for the access node. This process is identical to setting up a [single
node standby][single-ha].

The next step is to enable native replication on a distributed
hypertable. Native replication is governed by the
`replication_factor`, which determines how many data nodes a chunk is
replicated to. This setting is configured separately for each
hypertable, which means the same database can have some distributed
hypertables that are replicated and some that are not.

By default, the replication factor is set to `1`, so there is no
native replication. You can increase this number when you create the
hypertable. For example, to replicate the data across a total of three
data nodes:
```sql
SELECT create_distributed_hypertable('conditions', 'time', 'location',
	replication_factor => 3);
```

Alternatively, you can set the `replication_factor` on an existing

> hypertable in a multi-node environment using the

Reviewer: Turn this into a `<procedure>`

[`set_replication_factor`][set_replication_factor] call. Note,
however, that only new chunks will be replicated according to the
updated replication factor. Existing chunks need to be re-replicated
by copying those chunks to new data nodes (see the section below on
handling node failures).

When native replication is enabled, the replication happens whenever
you ingest data into the table. On each `INSERT` and `COPY` call, each
row of the data is written to multiple data nodes. This means that you
don't need to do any extra steps to have newly ingested data
replicated. When you query replicated data, the query planner only
includes one replica of each chunk in the query plan.

> ### Handling node failures

Reviewer: ```suggestion
### Node failures
```

WARNING:  distributed hypertable "conditions" is under-replicated
```

> If the data node cannot be recovered, native replication allows the
> node to be deleted from the multi-node cluster without losing
> data. Note that you must use the `force` option when deleting the data
> node if the deletion means that the cluster would no longer achieve
> the desired replication factor.

Reviewer: ```suggestion
If the data node cannot be recovered, native replication allows the
node to be deleted from the multi-node cluster without losing
data. Use the `force` option when you delete the data node if 
the deletion means that the cluster no longer achieves the 
desired replication factor.
```
